crates/starknet-server/src/api/json_rpc/spec_reader/spec_modifier.rs

```diff
@@ -114,10 +114,10 @@ impl SpecModifier {
             remove_array_element(&mut json_obj_spec, &path_parts, array_element_to_remove.index);
         }
 
-        // Parse the spec into a Spec struct
-        let spec: Spec =
-            serde_json::from_value(json_obj_spec).expect("Could not parse the JSON-RPC spec");
+        let json_spec_str = serde_json::to_string_pretty(&json_obj_spec)
+            .expect("could not serialize the spec to string");
 
-        spec
+        // Parse the spec into a Spec struct
+        serde_json::from_str(&json_spec_str).expect("Could not parse the JSON-RPC spec")
```

Comment:
Why is this now a two-step procedure (instead of single-step)?

Reply:
If there is an error when deserializing from serde_json::Value, it doesnt show on which line the error is. This is not the case with when deserializing from string. It shows on which line and column the deserialization failed

Reply:
cool, you can add a short comment explaining it

Reply:
[a74883a](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/257/commits/a74883a485835db6c81ec3ad890e70f766a27a86)